nan.h

```diff
@@ -1527,26 +1755,28 @@ class NanObjectWrap : public node::ObjectWrap {
 template <typename T, typename P>
 class NanWeakCallbackData {
  public:  // constructors
+  inline ~NanWeakCallbackData();
+
+ private:  // constructors
+  NAN_DISALLOW_ASSIGN_COPY_MOVE(NanWeakCallbackData)
+
+ public:
   typedef void (*Callback)(
       NanWeakCallbackData & data  // NOLINT(runtime/references)
       );
-  NanWeakCallbackData(v8::Handle<T> handle, P* param, Callback cb)
-    : parameter(param), callback(cb) {
-    NanAssignPersistent(persistent, handle);
-    Revive();
+  NanWeakCallbackData(v8::Persistent<T> *handle, P* param, Callback cb)
+    : persistent(), parameter(param), callback(cb) {
+    std::memcpy(&persistent, handle, sizeof(persistent));
```

Comment:
Meh, it just replicates the private Persistent constructor from T*. http://stackoverflow.com/a/3022238

On May 18, 2015 10:34:51 PM EEST, Ben Noordhuis <notifications@github.com> wrote:
>>    typedef void (*Callback)(
>>        NanWeakCallbackData & data  // NOLINT(runtime/references)
>>        );
>> -  NanWeakCallbackData(v8::Handle<T> handle, P* param, Callback cb)
>> -    : parameter(param), callback(cb) {
>> -    NanAssignPersistent(persistent, handle);
>> -    Revive();
>> +  NanWeakCallbackData(v8::Persistent<T> *handle, P* param, Callback
>cb)
>> +    : persistent(), parameter(param), callback(cb) {
>> +    std::memcpy(&persistent, handle, sizeof(persistent));
>
>That's not what 'trivial' means in this context. :-)  See here:
>http://en.cppreference.com/w/cpp/concept/TriviallyCopyable
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/nodejs/nan/pull/348/files#r30539839